File: Java/app/src/main/java/com/example/randomizermenumakanan/View/LeftFragment.java
package com.example.randomizermenumakanan.View;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.fragment.app.Fragment;

import com.example.randomizermenumakanan.R;

public class LeftFragment extends Fragment implements View.OnClickListener {

    private Button btnHome;
    private Button btnMenu;
    private Button btnExit;
    private FragmentListener listener;

    public LeftFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_left, container, false);
        this.btnHome = view.findViewById(R.id.btn_home);
        this.btnMenu = view.findViewById(R.id.btn_menu);
        this.btnExit = view.findViewById(R.id.btn_exit);
        this.btnHome.setOnClickListener(this);
        this.btnMenu.setOnClickListener(this);
        this.btnExit.setOnClickListener(this);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof FragmentListener) {
            this.listener = (FragmentListener) context;
        } else {
            throw new ClassCastException(context.toString() + " must implement FragmentListener");
        }
    }

    @Override
    public void onClick(View view) {
        if (view == this.btnHome) {
            this.listener.changePage(1);
        } else if (view == this.btnMenu) {
            this.listener.changePage(2);
        } else if (view == this.btnExit) {
            this.listener.closeApplication();
        }
    }
}

File: Kotlin/app/src/main/java/com/example/randomizermenumakanan/Fragment/LeftFragment.kt
package com.example.randomizermenumakanan.Fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.example.randomizermenumakanan.R

class LeftFragment : Fragment(), View.OnClickListener {
    private lateinit var btnHome: Button
    private lateinit var btnmenu: Button
    private lateinit var btnExit: Button
    private lateinit var listener: FragmentListener

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_left, container, false)
        this.btnHome = view.findViewById(R.id.btn_home)
        this.btnmenu = view.findViewById(R.id.btn_menu)
        this.btnExit = view.findViewById(R.id.btn_exit)
        this.btnHome.setOnClickListener(this)
        this.btnmenu.setOnClickListener(this)
        this.btnExit.setOnClickListener(this)
        return view
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is FragmentListener) {
            listener = context
        } else {
            throw ClassCastException("$context must implement FragmentListener")
        }
    }

    override fun onClick(view: View) {
        if (view === btnHome) {
            listener!!.changePage(1)
        } else if (view === btnmenu) {
            listener!!.changePage(2)
        } else if (view === btnExit) {
            listener!!.closeApplication()
        }
    }
}